### PR TITLE
#68 add isFeatureIncluded()

### PR DESCRIPTION
**Pull request by nlambert**

### CODE REVIEW

File: src/utils/is-feature-included/test/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import test from 'tape';
```
Comment (kennylavender) [inline src/utils/is-feature-included/test/index.js]: Eric would like us to use riteway for new tests. The package is already added, just convert the import and test ;)

**Comment by nlambert** [comment]
Riteway used :-)

File: src/utils/is-feature-included/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const isFeatureIncluded = (features = [], featureName = '') => features.includes(featureName);
+
+export default isFeatureIncluded;
```
Comment (kennylavender) [inline src/utils/is-feature-included/index.js]: Let's add ramda curry and a curry test

**Comment by nlambert** [comment]
added :-)

**Comment by nlambert** [comment]
Should we be merging this, or is there a higher authority? :-)

**Comment by kennylavender** [comment]
I left it for you to merge in because I am unsure if it meets the requirements of the issue. I thought maybe you and Eric discussed it and you would understand the requirements better.

If it meets the requirements of the issue its good to be merged in.

The reason I am unsure, is because the function currently is simply checking an array of strings for a specific string, `xs => s => xs.includes(s)`. I feel like the name of this should be more general for a utility of this nature and in that case probably does not belong in this package and it probably already in a general utility library like lodash or ramda.

To me a function named `isFeatureIncluded` seems more specific to this package and should probably take an array of Features that implement the Feature interface.

```javascript
interface Feature {
  name: String,
  enabled: false,
  dependencies?: [...featureName: String]
}
const features = [ ...Feature]
```

Thoughts?

**Comment by nlambert** [comment]
We noticed that in the client the `features` array being passed down through `context` only contained the name/ids of the features, and not all the feature info (like url). Just a flat list of strings.

**Comment by kennylavender** [comment]
Ok, what is the goal of `isFeatureIncluded`? What information are you trying to get with this function?

1. If the feature name is in the list of enabled features? 
2. If the feature name was provided original to withFeatures whether or not its disabled or enabled?
3. Something else?

I ask because, `withFeatures` takes all the features, then reduces it down to only the __enabled__ feature names and places that in context. Names of features that are __not__ enabled are not in the array.

If number 1, then the name of this function doesn't make much sense I don't think and I think it makes more sense to use ([contains](http://ramdajs.com/docs/#contains)) instead of creating a new function. `contains` does exactly what this function currently does, but also offers checking for composite objects as well.

**Comment by nlambert** [comment]
Thanks for that @kennylavender. It's probably # 1, but @ericelliott should chime in. I don't know this project well enough to provide valuable input.

**Comment by ericelliott** [comment]
I think we should provide a utility here rather than force users to use a generic `contains` or `includes`.

**Comment by kennylavender** [comment]
Ok, I think the name doesn't really fit, but I can't think of a better one so I am good with this if you guys are. @nlambert feel free to merge

**Comment by ericelliott** [comment]
@kennylavender I'm open to discussion about the name. The purpose of this function is:

* Given the client features, which are stripped of information for network performance reasons, determine whether or not a feature is enabled in the client.

The best reason to use a provided utility rather than push that responsibility to the users is that we may want to change the data  structure of the features in the future. If users have been checking with native JS or a third-party contains function, we're stuck always feeding the data through in the same format.

**Comment by kennylavender** [comment]
Yup, that makes sense. I see now how you came up with the name. 

I think a better solution would be to adjust the react context to hold the original feature data and then use the function we already have named `getIsEnabled` instead of creating a new function.

The only reason I original stripped down the features to an array of enabled names was because I was 'micro optimizing' when I most likely should not have. :)

**Comment by kennylavender** [comment]
@ericelliott @nlambert I am going to close this and #68 , and update the context to contain all of the feature data so we can use `getIsEnabled` and `getEnabled` from the client side like I described above.

Let me know if you disagree.

Here is the new issue #77

**Comment by kennylavender** [comment]
@nlambert we are going to go ahead with this function, sorry for delaying it :P 

Do you mind cleaning up the conflicts and moving this module out of the `source/utils` directory and just into `source` as I describe  in #70

**Comment by nlambert** [comment]
@kenny should be good to go :-)

**Comment by kennylavender** [comment]
@nlambert awesome thanks!

**Pull request merged**